designs/virtualization/4.5/Scheduling/CPU-features/cpu-features.md

```diff
@@ -0,0 +1,20 @@
+---
+parent: virtualization
+version: 4.5
+---
+
+# Fine Grained CPU Features
+
+Enabling fine grained features for VM.
+
+![VM page CPU features default](img/CPU-features-0-0.jpg)
+CPU Features is placed in the Scheduling section of the VM Details tab.
+
+![CPU features modal](img/CPU-features-1-0.jpg)
```

Comment:
Would this be a good opportunity to offer more info on what these features are?

Reply:
+1!

Reply:
un-resolving this comment. not sure what was changed but it still feels like we could provide further info around what each of these features are. WDYT?

Reply:
Just seeing the definitions above in this doc, should we also include this in the UI?

Reply:
My bad. Added info at the modal level and per each feature. the info is a lil thin thought. Asking Gilad if it will suffice for now
@glekner WDYT about the popovers in row 25 (image 1-1)

Reply:
ok just to clarify too these popover would be activated via an (i) with each checkbox?

Reply:
Yes. I think it's an underline now instead of (i). Did they change it back?

Reply:
These would be (i), dotted underlines are currently only used on Details pages where showing a ton of (i) icons became far too visually distracting. Forms should continue to use (i) which is what PatternFly still recommends.

Looking at the WIP info popovers below, if the descriptions stay that short I wonder if they could be included inline rather than within popovers. We really should ask technical stakeholders to explain what these features are in human people language though. 😄

I'm also not sure we link to upstream project documentation elsewhere in the Console, and URLs aren't typically shown raw - maybe "Learn more" with an external link icon if we know where it'll be within OpenShift's docs.

Example of dotted underlines:

![image](https://user-images.githubusercontent.com/9122899/78689434-53054980-78c4-11ea-9b35-86f2e42c403a.png)

Reply:
<img width="1680" alt="Screen Shot 2020-04-08 at 10 03 14" src="https://user-images.githubusercontent.com/48942365/78754462-30d10180-7980-11ea-878a-ff371f1b2613.png">
> I'm also not sure we link to upstream project documentation elsewhere in the Console, and URLs aren't typically shown raw - maybe "Learn more" with an external link icon if we know where it'll be within OpenShift's docs.

I also thought that until I saw a raw link to Kubernetes.io. within the console
Maybe it was a mistake. I'll be glad to fix it.

Reply:
Ah, great catch. It may be worth discussing this with the group/stakeholders again to see if opinions have changed. The Console pulls the text for those definition popovers from the [Kubernetes OpenAPI specification](https://github.com/kubernetes/kubernetes/tree/master/api/openapi-spec) which is written by the upstream community, and we started doing that a few months ago. Maybe that means we're okay with linking to other upstream project documentation too now, but internal docs would probably still be best when possible.

Although in this case, [the upstream documentation](https://kubevirt.io/api-reference/v0.26.0/definitions.html#_v1_features) we would be linking to doesn't have any more information than what's in this design. 😄 Maybe we should be asking the KubeVirt community to better describe what these features are first!